platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/tipset/TipsetAdvancementWeight.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.event.tipset;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Stores a weighed tipset advancement score.
+ * <p>
+ * If zero stake nodes were not a thing, we could use a long as a tipset advancement score. Or, if it was ok to ignore
+ * zero stake nodes, we could do the same as well. But since we don't want to allow zero sake nodes to get stale events,
+ * we need to have a mechanism for separately tracking when zero stake nodes have advanced.
+ *
+ * @param advancementWeight          the advancement weight provided by nodes with non-zero stake. Contributes to
+ *                                   meeting the threshold required to advance the current snapshot.
+ * @param zeroStakeAdvancementWeight the advancement weight provided by nodes with zero stake, this is incremented by
+ *                                   one for each zero stake node that advances. Does not contribute to meeting the
+ *                                   threshold required to advance the current snapshot.
+ */
+public record TipsetAdvancementWeight(long advancementWeight, long zeroStakeAdvancementWeight) {
```

Comment:
There are some mixing of terms here. The record is called `TipsetAdvancementWeight` but the javadoc says `weighed tipset advancement score`. The noun is different between the two: Weight vs. Score. Are they the same? If yes, then the terminology needs to be consistent. If they are different, the difference must be made clear.

Reply:
WeightedTipsetScore?

Reply:
They all mean the same thing, but it's not currently super consistent.

I think my preferred term is "tip advancement weight", or "adancement weight" for short.